api/base/filters.py

```diff
@@ -20,53 +28,128 @@ def filter_queryset(self, request, queryset, view):
+                if type(field) in self.TIME_FIELDS:
+                    time_match = self.DATETIME_PATTERN.match(value)
+                    if op != 'eq' or time_match:  # comparison or explicit time
+                        fields[field_name].append({
+                            'op': op,
+                            'value': self.convert_value(value, field)
+                        })
+                    else:  # match whole day; TODO: let times be as generic as possible (i.e. whole month, whole year)
+                        start = self.convert_value(value, field)
+                        stop = start + datetime.timedelta(days=1)
+                        fields[field_name].append({
+                            'op': 'gte',
+                            'value': start
+                        })
+                        fields[field_name].append({
+                            'op': 'lt',
+                            'value': stop
+                        })
+                else:
+                    fields[field_name].append({
+                        'op': op,
+                        'value': self.convert_value(value, field)
+                    })
+        return fields
+
+    def convert_key(self, field_name, field):
+        """Used so that that queries on fields with the souce attribute set will work
+        :param basestring field_name: text representation of the field name
+        :param rest_framework.fields.Field field: Field instance
+        """
+        return field.source or field_name
 
-    # Used to convert string values from query params to Python booleans when necessary
     def convert_value(self, value, field):
-        field_type = type(self.serializer_class._declared_fields[field])
-        value = value.strip()
+        """Used to convert string values from query params to bools and dates when necessary
+        :param basestring value: value to be resolved
+        :param rest_framework.fields.Field field: Field instance
+        """
+        field_type = type(field)
         if field_type == ser.BooleanField:
             if utils.is_truthy(value):
                 return True
             elif utils.is_falsy(value):
                 return False
-            # TODO Should we handle if the value is neither TRUTHY nor FALSY (first add test for how we'd expect it to
-            # work, then ensure that it works that way).
+            else:
+                raise InvalidFilterValue(
+                    value=value,
+                    field_type='bool'
+                )
+        elif field_type == ser.DateTimeField or field_type == ser.DateField:
+            try:
+                return date_parser.parse(value)
+            except ValueError:
+                raise InvalidFilterValue(
+                    value=value,
+                    field_type='date'
+                )
+        elif field_type in (ser.DecimalField, ser.IntegerField):
+            return float(value)
```

Comment:
I wonder if we should just use `field.to_internal_value` for all these conversions?